### ISSUE TITLE
Failling test on auto_reload

### ISSUE THREAD

**Issue by ph** [elastic/logstash#5160]
```
And another one, non metric related (seed was 40152)

  1) LogStash::Agent#execute when auto_reload is true when calling reload_state! with a config that does not contain reload incompatible plugins does not reload if new config contains reload incompatible plugins
     Failure/Error: expect(subject).to receive(:upgrade_pipeline).at_least(2).times
       (#<LogStash::Agent:0x278d2891>).upgrade_pipeline(any args)
           expected: at least 2 times with any arguments
           received: 1 time with any arguments
     # ./logstash-core/spec/logstash/agent_spec.rb:156:in `(root)'

```
@jsvd Want to take a look?

**Comment by jsvd** [comment]
thanks for reporting, checking. is this on master, or 2.x?

**Comment by ph** [comment]
@jsvd master, really random :(

**Comment by jsvd** [comment]
closed by #5175
